dask_sql/physical/rex/core/call.py

```diff
@@ -629,10 +629,21 @@ def extract(self, what, df: SeriesOrScalar):
             return df.week
         elif what in {"YEAR", "YEARS"}:
             return df.year
+        elif what == "DATE":
+            return df.date()
         else:
             raise NotImplementedError(f"Extraction of {what} is not (yet) implemented.")
 
 
+class ExtractDateOperation(Operation):
+    def __init__(self):
+        super().__init__(self.extract_date)
+
+    def extract_date(self, df: SeriesOrScalar):
+        df = convert_to_datetime(df)
+        return df.date
```

Comment:
since `date` is a special case in that returns a numpy array of python datetime.date objects instead of a standard numpy type do we know what would happen if we try using the results from an extract elsewhere within a query?

Reply:
It looks like it depends on the query. For example, the query
```
SELECT EXTRACT(YEAR FROM EXTRACT(DATE FROM d)) as "date" FROM df
```
fails, but the query
```
SELECT TIMESTAMPADD(YEAR, 1, EXTRACT(DATE FROM d)) FROM df
```
works fine and produces the expected output.

Reply:
Let me know if there's any other specific cases you'd like me to try out.